example/cluster-machine-template-multi-az.yaml

```diff
@@ -0,0 +1,242 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: cluster-api-multi-az
+  namespace: default
+  labels:
+    cni: "cluster-api-multi-az-crs-cni"
+    ccm: "cluster-api-multi-az-crs-ccm"
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: ["10.42.0.0/16"]
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: OscCluster
+    name:  cluster-api-multi-az
+    namespace: default
+  controlPlaneRef:
+    kind: KubeadmControlPlane
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    name: "cluster-api-multi-az-control-plane"
+    namespace: default
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: OscCluster
+metadata:
+  name:  cluster-api-multi-az
+  namespace: default
+spec:
+  network:
+    clusterName: cluster-api-multi-az
+    loadBalancer:
+      loadbalancername: capo-k8s
+      clusterName: cluster-api-multi-az
+    net:
+      clusterName: cluster-api-multi-az
+    internetService:
+      clusterName: cluster-api-multi-az
+    natService:
+      clusterName: cluster-api-multi-az
+    bastion:
+      clusterName: cluster-api-multi-az
+      enable: false
+    subnets:
+    - name: cluster-api-multi-az-eu-west-2a
+      ipSubnetRange: ["10.42.0.0/24"]
+      subregionName: eu-west-2a
+    - name: cluster-api-multi-az-eu-west-2b
+      ipSubnetRange: ["10.42.1.0/24"]
+      subregionName: eu-west-2b
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  name: "cluster-api-multi-az-md-0-eu-west-2a"
```

Comment:
```
[manager] {"level":"error","ts":"2023-10-25T15:09:04Z","msg":"Reconciler error","controller":"osccluster","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"OscCluster","OscCluster":{"name":"cluster-api-multi-az","namespace":"default"},"namespace":"default","name":"cluster-api-multi-az","reconcileID":"a1b1cd98-fd3f-4a82-934a-3bed76bae96e","error":"cluster-api-multi-az-subnet-kw-5244ac4a-5d47-4a0e-9549-8fa762843acb subnet does not exist in routeTable
```

Reply:
You should set have  your routetable and  your sg 
Default rtb does match your subnet.
And default sg will not cover your ip range :)

Reply:
Yes good call, that made me think of something. I will have to have multiple NAT services as well, one for each subregion right? So i should also implement the possibility for creating multiple NAT services (Noz only one is supported). Tell me if I am missing something?

Reply:
You can have one natgateway but in order to be more resilient it is better to have one natgateway by availability zone.
You can implement multi natservice (https://docs.outscale.com/fr/userguide/%C3%80-propos-des-NAT-services.html) to have a new object natServices :     
```yaml
natServices:
- name: cluster-api-natservice-zonea
   clusterName: cluster-api
   publicipname: cluster-api-publicip-nat-zonea
   subnetname: cluster-api-subnet-public-zonea
- name: cluster-api-natservice-zoneb
   clusterName: cluster-api
   publicipname: cluster-api-publicip-nat-zoneb
   subnetname: cluster-api-subnet-public-zoneb     
```

In order to keep a compatibility with  cluster-api-provider-outscale you have to use natservices instead of natservice only if natServices are defined in osccluster. 
You can copy/paste natService and switch to an array of parameters like in this PR https://github.com/outscale/cluster-api-provider-outscale/pull/171/files

Reply:
All right, I'm on it!

Reply:
Ok done, we tested the newly created example and it works.
One thing we still need to do is manage the controlplanes so that they are deployed on multiple zones. However we are gonna do that in a different PR as this one is already pretty big.